code/modules/mob/pulling.dm

```diff
@@ -15,6 +15,9 @@
 				return
 			stop_pulling()
 
+		if(ispet(src) && isliving(AM) && !ispet(AM))
```

Comment:
This is no more acceptable than the last, `istype(src)`. Override pet/start_pulling.